File: healthcare_pro/healthcare_management/web_form/patient_portal/patient_portal.py
import frappe


def get_context(context):

    def on_submit(doc):
        patient_conflict = frappe.db.exists("Healthcare Patient", {
            "full_name": doc.full_name,
            "date_of_birth": doc.date_of_birth,
            "gender": doc.gender,
            "mobile": doc.mobile,
            "email": doc.email,
            "name": ["!", doc.name],
        })
        if patient_conflict:
            frappe.throw("Patient already exists")


    context.on_submit = on_submit
    # Button href is upgraded with ?patient=<name> by client script after save.
    context.success_message = (
        "<div class=\"alert alert-success\">You have been successfully registered. Do you wish to book an appointment?</div>"
        "<a class=\"btn btn-primary book-appointment-btn\" href=\"/book-an-appointment\">Book an Appointment</a>"
    )

    return context

File: healthcare_pro/healthcare_management/web_form/patient_portal/patient_portal.js
frappe.ready(function() {
    if (frappe.web_form) {
        frappe.web_form.after_save = function(data) {
            var patient = (data && (data.patient || data.name))
                || (frappe.web_form.doc && frappe.web_form.doc.name);

            if (patient) {
                try {
                    localStorage.setItem('book_appointment_patient', patient);
                } catch (e) {
                    // Ignore storage errors and continue with URL-based flow.
                }

                var targetUrl = '/book-an-appointment/new?patient=' + encodeURIComponent(patient);
                var btn = document.querySelector('.book-appointment-btn')
                    || document.querySelector('.success-state a.btn-primary');

                if (btn) {
                    btn.href = targetUrl;
                }

                var button = document.querySelector('.success-state .btn-primary');
                if (button && button.tagName === 'BUTTON') {
                    button.onclick = function() {
                        window.location.href = targetUrl;
                    };
                }
            }
        };
    }
});
